standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java

```diff
@@ -769,6 +769,28 @@ public List<Partition> run(List<Long> input) throws MetaException {
     });
   }
 
+  public List<Partition> getPartitionsViaSqlPs(Table table, GetPartitionsArgs args) throws MetaException {
+    String catName = table.getCatName();
+    String dbName = table.getDbName();
+    String tblName = table.getTableName();
+
+    String sqlFilter = "" + PARTITIONS + ".\"PART_NAME\" like ? ";
+    String partialName = MetaStoreUtils.makePartNameMatcher(table, args.getPart_vals(), "_%");
```

Comment:
have you tested the matcher against other db? such as Postgres, MySQL, etc.

Reply:
it has been used in: https://github.com/apache/hive/blob/978194b7739e53b4b0e555a8b616a1ccf7d8d437/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java#L825-L828